monopod/pkg/commands/scaffold.go

```diff
@@ -72,47 +78,86 @@ func (o scaffoldOptions) runScaffold(ctx context.Context) error {
 		return err
 	}
 
-	modroot := filepath.Join(o.OutputPath, "images", o.ModuleName)
+	if err := o.mkImageDir(); err != nil {
+		return err
+	}
 
-	if err := os.MkdirAll(modroot, os.ModePerm); err != nil {
-		return fmt.Errorf("unable to generate target folder images/%s: %w", o.ModuleName, err)
+	if err := filepath.Walk("images/TEMPLATE", o.walk); err != nil {
+		return err
 	}
 
-	if err := filepath.Walk("images/TEMPLATE", func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
+	// create readme hcl metadata file
+	nr := newReadmeImpl{
+		Readme:      completeReadme{Name: o.PackageName},
+		renderedHCl: new(bytes.Buffer),
+	}
+	if err := nr.render(); err != nil {
+		return err
+	}
+	if err := nr.write(); err != nil {
+		return err
+	}
 
-		if info.IsDir() {
-			return nil
-		}
-		repl := strings.Replace(path, "TEMPLATE", o.ModuleName, 1) // Replacing TEMPLATE with the module name
-		if err := os.MkdirAll(filepath.Dir(repl), os.ModePerm); err != nil {
-			return err
-		}
+	// render the readme's markdown
+	rr := NewReadmeRenderer(o.PackageName)
+	rr.Readme = &nr.Readme
```

Comment:
Just some styling nits, feel free to ignore:

- why create a new renderer, then assign its `Readme` -- is this something that could be passed to the constructor?
- `NewReadmeRenderer` doesn't need to be exported

Reply:
Great suggestion - I've reworked the constructor to accept one, which will hopefully be a bit less confusing for future travelers. It is a bit of an odd use case where I didn't want to go through all the same calls in `readme_render.go` to read, validate, populate, render, and write out stuff.